x-pack/plugins/apm/server/lib/correlations/get_correlations_for_failed_transactions/index.ts

```diff
@@ -162,32 +160,25 @@ export async function getErrorRateTimeSeries({
     body: {
       size: 0,
       query: { bool: { filter: backgroundFilters } },
-      aggs: {
-        // overall aggs
-        timeseries: timeseriesAgg,
-
-        // per term aggs
-        ...perTermAggs,
-      },
+      aggs: merge({ timeseries: timeseriesAgg }, perTermAggs),
```

Comment:
Does `perTermAggs` contains a `timeseries` key that needs to be deeply merged or why is spread not sufficient?

Reply:
The spread operator ignores `{ [x:string]: ... }` when merging the type, whereas `merge` creates an intersection type that keeps string-indexed types. That allows us to get rid of the `@ts-expect-error` directive.

Reply:
Okay. I prefer spread where possible because it's more limited in scope but I see the issues with the types.